File: Java/HashSet/Keep Multiplying Found Values by Two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        HashSet<Integer> st=new HashSet<>();
        for(int num:nums)        {
            st.add(num);
        }
        int tar=original;
        while(st.contains(tar)){
            tar*=2;
        }
        return tar;
    }
}

File: CPP/Set/Keep Multiplying Found Values by Two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        set<int> st;
        for (auto num : nums) {
            st.insert(num);
        }
        int target = original;
        while (st.count(target)) {
            target = 2 * target;
        }
        return target;
    }
};
